library/src/main/java/com/climate/mirage/Mirage.java

```diff
@@ -165,14 +166,18 @@ public MirageRequest load(String uri) {
 	 */
 	public MirageRequest load(Uri uri) {
 		MirageRequest r = requestObjectPool.getObject();
-		if (uri.getScheme().startsWith(SCHEME_FILE) ||
-                uri.getScheme().startsWith(SCHEME_ANDROID_RESOURCE) ) {
-			r.diskCacheStrategy(DiskCacheStrategy.RESULT);
+		String scheme = uri != null ? uri.getScheme() : null;
+		if (!TextUtils.isEmpty(scheme)) {
+			if (scheme.startsWith(SCHEME_FILE) ||
+					scheme.startsWith(SCHEME_ANDROID_RESOURCE)) {
+				r.diskCacheStrategy(DiskCacheStrategy.RESULT);
+			}
 		}
 		return r.mirage(this).uri(uri);
 	}
 
 	public MirageRequest load(File file) {
```

Comment:
Looks good, thoughts about moving this load method next to the other one so the overloaded methods are in sequence?